packages/block-library/src/classic/edit.js

```diff
@@ -84,12 +92,29 @@ export default class ClassicEdit extends Component {
 		}
 
 		editor.on( 'blur', () => {
+			editor.wpClassicBlockBookmark = editor.selection.getBookmark();
+
+			// Keep the "raw" content as it contains the caret position bookmark.
 			setAttributes( {
+				contentRaw: editor.getContent( { format: 'raw' } ),
 				content: editor.getContent(),
 			} );
 			return false;
 		} );
 
+		editor.on( 'init', () => {
+			// Store the initial "raw" content.
+			setAttributes( {
+				contentRaw: editor.getContent( { format: 'raw' } ),
```

Comment:
I wouldn't expect merely initialising the block to change the content. The user didn't do  anything.